ui/src/app/shared/services/applications-service.ts

```diff
@@ -20,30 +23,54 @@ function optionsToSearch(options?: QueryOptions) {
     return {};
 }
 
+function getQuery(projects: string[], isFromApps: boolean, options?: QueryOptions): any {
+    if (isFromApps) {
+        return {projects, ...optionsToSearch(options)};
+    } else {
+        return {...optionsToSearch(options)};
+    }
+}
+
 export class ApplicationsService {
-    public list(projects: string[], options?: QueryOptions): Promise<models.ApplicationList> {
+    public list(
+        projects: string[],
+        ctx: ContextApis & {
+            history: History<unknown>;
+        },
+        options?: QueryOptions
+    ): Promise<models.AbstractApplicationList> {
+        const isFromApps = isInvokedFromAppsPath(ctx.history.location.pathname);
         return requests
-            .get('/applications')
-            .query({projects, ...optionsToSearch(options)})
-            .then(res => res.body as models.ApplicationList)
+            .get(getRootPathByPath(ctx.history.location.pathname))
+            .query(getQuery(projects, isFromApps, options))
```

Comment:
For getQuery, could we just return the original in both cases?  eg ` .query({projects, ...optionsToSearch(options)})`